File: ui/components/ui/ProducerSession/message-generator/sequence-id-generator/sequence-id-generator.ts
import * as pb from '../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';
import { Int64Generator, int64GeneratorFromPb, int64GeneratorToPb } from '../data-generators/int64/int64-generator';

export type SequenceIdGenerator = {
  type: 'sequence-id-generator',
  generator: Int64Generator
};

export function sequenceIdGeneratorFromPb(v: pb.SequenceIdGenerator): SequenceIdGenerator {
  return {
    type: 'sequence-id-generator',
    generator: int64GeneratorFromPb(v.getGenerator()!)
  };
}

export function sequenceIdGeneratorToPb(v: SequenceIdGenerator): pb.SequenceIdGenerator {
  const generatorPb = new pb.SequenceIdGenerator();
  generatorPb.setGenerator(int64GeneratorToPb(v.generator));

  return generatorPb;
}

File: server/src/main/scala/producer/message_generator/sequence_id_generator/SequenceIdGenerator.scala
package producer.message_generator.sequence_id_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import producer.message_generator.data_generators.int64.Int64Generator

case class SequenceIdGenerator(
    generator: Int64Generator
)

object SequenceIdGenerator:
    def fromPb(v: pb.SequenceIdGenerator): SequenceIdGenerator =
        SequenceIdGenerator(
            generator = Int64Generator.fromPb(v.generator.get)
        )

    def toPb(v: SequenceIdGenerator): pb.SequenceIdGenerator =
        pb.SequenceIdGenerator(
           generator = Some(Int64Generator.toPb(v.generator))
        )
